server-plugins/tracker-resources/src/index.ts

```diff
@@ -336,6 +336,28 @@ async function doIssueUpdate (
       }))
     )
   }
+
+  if (Object.prototype.hasOwnProperty.call(updateTx.operations, 'sprint')) {
+    if (updateTx.operations.sprint !== null) {
+      const [sprint] = await control.findAll(
+        tracker.class.Sprint,
+        { _id: updateTx.operations.sprint as Ref<Sprint> },
+        { limit: 1 }
+      )
+      res.push(
+        control.txFactory.createTxUpdateDoc(updateTx.objectClass, updateTx.objectSpace, updateTx.objectId, {
+          project: sprint.project
+        })
+      )
+    } else {
+      res.push(
+        control.txFactory.createTxUpdateDoc(updateTx.objectClass, updateTx.objectSpace, updateTx.objectId, {
+          project: null
```

Comment:
Why you remove project for any non sprint change operation?

Reply:
`if (Object.prototype.hasOwnProperty.call(updateTx.operations, 'sprint'))` prevents this, so we update project this way only if the sprint was changed

Reply:
In case user sets "No sprint". So that it sets both sprint and project to null.
Is it supposed to set project only when the user sets any existing sprint or when it's set to "No sprint" too?